src/resources/js/modules/enso/__.js

```diff
@@ -0,0 +1,30 @@
+import store from '../../store';
+
+const __ = store.getters[ 'localisation/__' ];
+
+export default function (key, params) {
+    if (!store.getters[ 'localisation/isInitialised' ]) {
+        return key;
+    }
+
+    let translation = __(key);
+
+    if (typeof translation === 'undefined'
+        && store.state.localisation.keyCollector) {
+        store.dispatch('localisation/addMissingKey', key);
+    }
+
+    translation = translation || key;
+    if(params) {
+        translation = translation.replace(/:(\w*)/g, function(e, key) {
+            let param = params[key.toLowerCase()] || key;
+            if(key === key.toUpperCase()) { // param is uppercased
+                param = param.toUpperCase();
+            } else if(key[0] === key[0].toUpperCase()) { // first letter is uppercased
+                param = param.charAt(0).toUpperCase() + param.slice(1);
+            }
+            return param;
+        });
+    }
+    return translation;
+}
```

Comment:
I did a small refactor that covers some edge cases, but please test it (I suppose you have everything set for that)

```vue
import store from '../../store';

export default (key, params = null) => {
    if (!store.getters['localisation/isInitialised']) {
        return key;
    }

    let translation = store.getters['localisation/__'](key);

    if (typeof translation === 'undefined') {
        translation = key;

        if (store.state.localisation.keyCollector) {
           store.dispatch('localisation/addMissingKey', key);
        }
    }

    return !!params && typeof params === 'object'
        ? translation.replace(/:(\w*)/g, (e, key) => {
            if (typeof params[key.toLowerCase()] === 'undefined') {
                return key;
            }

            const param = params[key.toLowerCase()];

            if(key === key.toUpperCase()) {
                return param.toUpperCase();
            }
            
            return key[0] === key[0].toUpperCase()
                ? param.charAt(0).toUpperCase() + param.slice(1)
                : param;
            }
        })
        : translation;
}
```

Reply:
Had to fix a syntax error & add a `== null` check to translation (missing keys are returning a null object instead of undefined), committing changes